app.py

```diff
@@ -39,71 +39,72 @@ def calendar(season: str):
-
-    best_n_col = str(df.filter(regex=r"Best.*").columns[0])
-    n = best_n_col.split("-")[0][4:]
-    df = (
-        df.rename(
-            columns={
-                best_n_col: f"Součet ({n} z {len(oris_ids_in_results)})",
-                "Name": "Jméno",
-            }
+            # Fix formatting
+            for place_col in df.filter(regex=r".*-Place"):
+                if df[place_col].dtype == float:
+                    df[place_col] = df[place_col].apply(lambda x: f"{x:.0f}.")
+            for points_col in df.filter(regex=r".*-Points"):
+                if df[points_col].dtype == float:
+                    df[points_col] = df[points_col].apply(
+                        lambda x: f"{x:.0f}"
+                    )  # contains nans -> can't be casted to int
+            category_dfs.append(df)
+        df = pd.concat(category_dfs)
+
+        # Process DataFrame (rename and drop columns etc.)
+        events = em.get_all_events(season)
+        if events is None:
+            return render_template("results.html", season=season, results={})
+        oris_id_to_name_mapping = {
+            ev.oris_id: ev.name for ev in events.values() if ev.oris_id
+        }
+
+        oris_ids_in_results = set(
+            [int(x.split("-")[0]) for x in df.columns if x[0].isdigit()]
         )
-        .replace(["nan (nan.)", "nan (nan)"], "---")
-        .drop(columns=cols_to_drop)
-    )
-
-    # Split DataFrame per category
-    results = {}
-    for category, group in df.groupby("category"):
-        group_df = group.set_index("place", drop=True)
-        results[category] = group_df.drop(columns=["category"])
 
-    return render_template("results.html", season=season, results=results)
+        cols_to_drop = []
+        for oris_id in oris_id_to_name_mapping.keys():
+            if oris_id in oris_ids_in_results:
+                df[oris_id_to_name_mapping[oris_id]] = (
+                    df[f"{oris_id}-Points"].astype(str)
+                    + " ("
+                    + df[f"{oris_id}-Place"].astype(str)
+                    + ")"
+                )
+                cols_to_drop.extend([f"{oris_id}-Points", f"{oris_id}-Place"])
+
+        best_n_col = str(df.filter(regex=r"Best.*").columns[0])
+        n = best_n_col.split("-")[0][4:]
+        df = (
+            df.rename(
+                columns={
+                    best_n_col: f"Součet ({n} z {len(oris_ids_in_results)})",
+                    "Name": "Jméno",
+                }
+            )
+            .replace(["nan (nan.)", "nan (nan)"], "---")
+            .drop(columns=cols_to_drop)
+        )
+        # Split DataFrame per category
+        for category, group in df.groupby("category"):
+            group_df = group.set_index("place", drop=True)
+            results[category] = group_df.drop(columns=["category"])
+    finally:
+        return render_template("results.html", seasons=seasons, season=season, results=results)
```

Comment:
Tady to je nejak zvlastni ten diff. Vidim spravne, ze tvoje zmeny jsou jen to `try-finally` (bez `catch`?) a vsechno uvnitr je stejne?

Reply:
ano, jsem si rikal ze kdyz se neco pokazi, tak se aspon neukaze error   
je to dirty, ale tak funguje to